src/policy/fees.cpp

```diff
@@ -603,36 +603,39 @@ void CBlockPolicyEstimator::processTransaction(const CTransactionRef& tx, const
     assert(bucketIndex == bucketIndex3);
 }
 
-bool CBlockPolicyEstimator::processBlockTx(unsigned int nBlockHeight, const CTxMemPoolEntry* entry)
+bool CBlockPolicyEstimator::processBlockTx(unsigned int nBlockHeight, const CTransactionRef& tx)
 {
     AssertLockHeld(m_cs_fee_estimator);
-    if (!_removeTx(entry->GetTx().GetHash(), true)) {
+    std::map<uint256, TxStatsInfo>::iterator pos = mapMemPoolTxs.find(tx->GetHash());
+    if (pos == mapMemPoolTxs.end()) {
         // This transaction wasn't being tracked for fee estimation
         return false;
     }
 
     // How many blocks did it take for miners to include this transaction?
     // blocksToConfirm is 1-based, so a transaction included in the earliest
     // possible block has confirmation count of 1
-    int blocksToConfirm = nBlockHeight - entry->GetHeight();
+    int blocksToConfirm = nBlockHeight - pos->second.blockHeight;
+
+    // Feerates are stored and reported as BTC-per-kb:
+    CAmount feerate = pos->second.m_fee_per_k;
+    _removeTx(tx->GetHash(), true);
```

Comment:
Why is this line moved?

Reply:
Because `_removeTx` clear the tx from `mapMemPoolTxs`.
So using
```c++
 _removeTx(tx->GetHash(), true);
 std::map<uint256, TxStatsInfo>::iterator pos = mapMemPoolTxs.find(tx->GetHash());
```  
`pos` will always be an empty iterator.

So I find the iterator of the tx from `mapMemPoolTxs` and capture all the values I need (tx fee rate and entry height) before removing the tx. 

Before this pull request, we could get all those details from CTxMempoolEntry.

Reply:
Thanks for the explainer :)
Feel free to resolve.